### ISSUE TITLE
Write an overview of liquid-fixpoint

### ISSUE THREAD

**Issue by facundominguez** [ucsd-progsys/liquid-fixpoint#543]
The README says this much

> This package implements a Horn-Clause/Logical Implication constraint solver used for various Liquid Types. The solver uses SMTLIB2 to implement an algorithm similar to: ...

which I think is not quite sufficient to get people excited about it. liquid-fixpoint caught me only after I learned about the PLE algorithm, and then REST, I do not understand elim as well, but I regard it essential to deal with abstract predicates in LH and inference of refinement predicates in general.

When I explain liquid-fixpoint to someone, I say it does three things:
1) It infers the refinement types of your program
2) It does the unfolding of functions in your constraints to prove them
3) It does the rewriting of your expressions using lemmas in the environment

I think these points, together with pointers to larger explanations, would be more compelling.

Ideally, I'd like to have a first defense of blogposts giving overviews for each algorithm, followed by a second defense with the academic publications. We already have the publications. Perhaps we have some of the blogposts too and we just need to put it all together, or perhaps there is some more writing to do.

**Comment by ranjitjhala** [comment]
Yes it’s definitely “dry” — pretty sure we have blog post for all of those
… let me try to dig up!
